sessiontxn/interface.go

```diff
@@ -55,6 +55,44 @@ type EnterNewTxnRequest struct {
 	StaleReadTS uint64
 }
 
+// StmtErrorHandlePoint is where the error is being handled
+type StmtErrorHandlePoint int
+
+const (
+	// StmtErrAfterQuery means we are handling an error after the query failed
+	StmtErrAfterQuery StmtErrorHandlePoint = iota
+	// StmtErrAfterLock means we are handling an error after lock failed.
+	StmtErrAfterLock
+)
+
+// StmtErrorAction is the next action advice when an error occurs when executing a statement
+type StmtErrorAction int
+
+const (
+	// StmtActionError means the error should be returned directly without any retry
+	StmtActionError StmtErrorAction = iota
+	// StmtActionRetryReady means the error is caused by this component, and it is ready for retry.
+	StmtActionRetryReady
+	// StmtActionNoIdea means the error is not caused by this component,
+	// and whether retry or not should be determined by other components.
+	StmtActionNoIdea
```

Comment:
May give some advice on what the user should do if this action is in the result for example returning the original error again.